Bookish/Bookish.Web/Controllers/HomeController.cs

```diff
@@ -56,13 +63,22 @@ public IActionResult AddBookPost(string title, string authors, string isbn, int
         {
             var isbnError = libraryService.BookExists(isbn);
             var copyError = copies < 1;
-            if (isbnError || copyError)
+            var invalidFormat = isbn?.Replace("-", "").Length != 13;
+            if (isbnError || copyError || invalidFormat)
             {
                 return RedirectToAction("AddBook", new { isbn, title, authors, copies });
             }
             
             libraryService.AddBook(isbn, title, authors, copies);
-            return RedirectToAction("Catalogue");
+            return RedirectToAction("AddedBook", new { isbn, copies });
+        }
+
+        public IActionResult AddedBook(string isbn, int copies)
+        {
+            var title = libraryService.GetBook(isbn).title;
+            var imageFolder = Path.Combine(webEnvironment.WebRootPath, "images", "barcodes");
```

Comment:
Any reason why the path can't be calculated in the barcode service? 🙂

Reply:
Is it fine to use the webEnvironment stuff in barcode service? I wasn't sure if it was just a controller thing

Reply:
Good point. I'd suggest trying it and see if it works 🙂